src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -306,22 +302,78 @@ ucs_status_t uct_ib_mlx5_devx_modify_qp_state(uct_ib_mlx5_qp_t *qp,
 
 void uct_ib_mlx5_devx_destroy_qp(uct_ib_mlx5_md_t *md, uct_ib_mlx5_qp_t *qp)
 {
-    int ret = mlx5dv_devx_obj_destroy(qp->devx.obj);
-    if (ret) {
-        ucs_error("mlx5dv_devx_obj_destroy(QP) failed: %m");
-    }
+    uct_ib_mlx5_devx_obj_destroy(qp->devx.obj, "QP", UCS_LOG_LEVEL_ERROR);
     uct_ib_mlx5_put_dbrec(qp->devx.dbrec);
     uct_ib_mlx5_md_buf_free(md, qp->devx.wq_buf, &qp->devx.mem);
 }
 
+ucs_status_t uct_ib_mlx5_devx_obj_modify(struct mlx5dv_devx_obj *obj,
+                                         const void *in, size_t inlen,
+                                         void *out, size_t outlen,
+                                         char *msg_arg, unsigned *syndrome_ptr)
+{
+    int ret = mlx5dv_devx_obj_modify(obj, in, inlen, out, outlen);
+    if (ret) {
+        ucs_error("mlx5dv_devx_obj_modify(%s) failed, syndrome %x: %m",
+                   msg_arg, *syndrome_ptr);
+        return UCS_ERR_IO_ERROR;
+    }
+    return UCS_OK;
+}
+
+struct mlx5dv_devx_obj*
+uct_ib_mlx5_devx_obj_create(struct ibv_context *context,
+                            const void *in, size_t inlen,
+                            void *out, size_t outlen,
+                            char *msg_arg, unsigned *syndrome_ptr)
+{
+    struct mlx5dv_devx_obj *obj = mlx5dv_devx_obj_create(context, in, inlen,
+                                                         out, outlen);
+    if (obj == NULL) {
+        ucs_error("mlx5dv_devx_obj_create(%s) failed, syndrome %x: %m",
+                   msg_arg, *syndrome_ptr);
+    }
+    return obj;
+}
+
+ucs_status_t uct_ib_mlx5_devx_obj_destroy(struct mlx5dv_devx_obj *obj,
+                                          char *msg_arg,
+                                          ucs_log_level_t log_level)
```

Comment:
maybe better `int silent` instead of `log_level` and interpret it internally like [here](https://github.com/openucx/ucx/blob/95be9aeb501858a653fa82b8f5a52f8b530f64c2/src/uct/ib/mlx5/ib_mlx5.h#L751)